src/types/index.d.ts

```diff
@@ -85,10 +90,8 @@ declare namespace State {
     lockedUntilBlockTimestamp: number
     interest: string
     daoTypeHash: string
-    udtInfo: UDTInfo
     cellInfo: CellInfo
-    mNftInfo: NftIssuer | NftClass | NftToken
-    nrc721TokenInfo: Record<'amount' | 'symbol', string>
+    extraInfo: UDTInfo | NftIssuer | NftClass | NftToken | nrc721TokenInfo
```

Comment:
nrc721TokenInfo -> Nrc721TokenInfo

In TypeScript, identifiers that do not exist in the `.d.ts` files will not trigger warnings. This can silently change their type to 'any' and propagate it throughout the codebase, so caution is advised.

`.d.ts` files have additional drawbacks. They should not be used as internal type definition files, and it is recommended to consider refactoring them into regular `.ts` files in the future.

Reply:
I suggest refactoring it into the following form to achieve type inference.
Example:
```typescript
interface Cell$Base {
  id: number
  addressHash: string
  capacity: string
  fromCellbase: boolean
  generatedTxHash: string
  targetBlockNumber: number
  baseReward: string
  secondaryReward: string
  commitReward: string
  proposalReward: string
  consumedTxHash: string
  status: 'live' | 'dead'
  isGenesisOutput: boolean
  cellIndex: string
  compensationStartedBlockNumber: number
  compensationEndedBlockNumber: number
  compensationStartedTimestamp: number
  compensationEndedTimestamp: number
  lockedUntilBlockNumber: number
  lockedUntilBlockTimestamp: number
  interest: string
  daoTypeHash: string
  cellInfo: CellInfo
  since?: {
    raw: string
    median_timestamp?: string
  }
}

interface Cell$NoExtra extends Cell$Base {
  cellType: 'normal' | 'nervos_dao_withdrawing'
}

interface Cell$UDT extends Cell$Base {
  cellType: 'udt'
  extraInfo: UDTInfo
}

interface Cell$NftIssuer extends Cell$Base {
  cellType: 'm_nft_issuer'
  extraInfo: NftIssuer
}

type Cell = Cell$NoExtra | Cell$UDT | Cell$NftIssuer
```
```typescript
if (cell.cellType === 'm_nft_issuer') {
  const nftInfo = cell.extraInfo // the type is State.NftIssuer
}
```

Reply:
It looks fixed by [3dc0c5](https://github.com/Magickbase/ckb-explorer-frontend/pull/6/commits/03dc0c5e8d695ae453ca9224ba3b5074432b2208) and [c68d620](https://github.com/Magickbase/ckb-explorer-frontend/pull/6/commits/c68d62084cce944e02b243e4021e13cdd1817414)

Reply:
> It looks fixed by [3dc0c5](https://github.com/Magickbase/ckb-explorer-frontend/pull/6/commits/03dc0c5e8d695ae453ca9224ba3b5074432b2208) and [c68d620](https://github.com/Magickbase/ckb-explorer-frontend/pull/6/commits/c68d62084cce944e02b243e4021e13cdd1817414)

yes

Reply:
I have made the necessary fixes as per the requirements. Here is the link: https://github.com/Magickbase/ckb-explorer-frontend/pull/6/commits/e2746e129b17a328c96ae145a2f5148cb6e6020a